hypertrace-gradle-java-convention-plugin/src/main/java/org/hypertrace/gradle/java/convention/JavaConventionPlugin.java

```diff
@@ -32,8 +31,8 @@ private void createConvention(Project target) {
   }
 
   private void configureCompatibility(Project target) {
-    Property<JavaVersion> javaVersion = javaConventionExtension(target).compatibilityVersion;
-    javaPluginExtension(target).setTargetCompatibility(javaVersion);
+    JavaVersion javaVersion = javaConventionExtension(target).compatibilityVersion.get();
```

Comment:
calling .get() here defeats the point of using deferred configuration - you want to assign the provider, not the value in it

Reply:
`setSourceCompatibility` with a property was not working

Reply:
Let us continue this on https://github.com/hypertrace/hypertrace-gradle-java-convention-plugin/pull/4